### PR TITLE
Refactor edit

### PR DESCRIPTION
**Pull request by Mamtha3005**

### CODE REVIEW

**Comment by Mamtha3005** [comment]
@marcusngwj

File: src/main/java/seedu/taskell/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -115,21 +100,34 @@ private boolean isValidDate(ReadOnlyTask taskToEdit) {
...
     private boolean isValidTime(ReadOnlyTask taskToEdit) {
+        TaskTime currentTime = TaskTime.getTimeNow();
         if (taskToEdit.getTaskType().equals(Task.EVENT_TASK)) {
             if (endDate.equals(startDate) && endTime.isBefore(startTime)) {
-                return false;
+                try {
+                    endDate = endDate.getNextDay();
+                } catch (IllegalValueException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/logic/commands/EditCommand.java]: Please handle the exception

File: src/main/java/seedu/taskell/logic/parser/Parser.java [hunk 2]
```diff
@@ -212,9 +208,14 @@ private Command prepareListDate(String arguments) {
 
         if (st.hasMoreTokens() || !TaskDate.isValidDate(date)) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListDateCommand.MESSAGE_USAGE));
-        }
+        } else {
+            try {
+                return new ListDateCommand(new TaskDate(date));
+            } catch (IllegalValueException ive) {
+                return new IncorrectCommand(ive.getMessage());
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/logic/parser/Parser.java]: Remove unnecessary newline

File: src/main/java/seedu/taskell/logic/parser/Parser.java [hunk 3]
```diff
@@ -434,56 +400,60 @@ private Command prepareAdd(String args) {
...
-        
-        addReservedWordToDescription(); //Add trailing reserved word to description
-        
+
+        addReservedWordToDescription(); // Add trailing reserved word to
+                                        // description
+
         extractDescriptionComponent();
-//        adjustEndDate();
+        // adjustEndDate();
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/logic/parser/Parser.java]: Remove it if no longer in use

**Comment by Mamtha3005** [comment]
Docuentation Reviewed

**Comment by Mamtha3005** [comment]
documentation Reviewed

**Pull request merged**